File: Binary_Search /python/10815.py
n = int(input())
arr = sorted(map(int,input().split()))
m = int(input())
target = list(map(int,input().split()))

def search (min_num,max_num,target_num):
    if min_num >= max_num: return "0"
    mid_num = min_num + (max_num-min_num)//2
    if arr[mid_num] == target_num : return "1"
    if arr[mid_num] > target_num : return search(min_num,mid_num,target_num)
    return search(mid_num+1,max_num,target_num)
result = []
for i in range(len(target)):
    result.append(search(0,len(arr),target[i]))
print(" ".join(result))

File: Binary_Search /javascript/10815.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');

const card = input[1].split(' ').map(Number).sort((a,b)=>a-b);
const targetArr = input[3].split(' ').map(Number);

const search = (min,max,targetNum) => {
  if (min>=max) return 0
  const mid = min + Math.floor((max-min)/2);
  if (card[mid] === targetNum) return 1
  if (card[mid] > targetNum) return search(min,mid,targetNum);
  else return search(mid+1,max,targetNum)
}
console.log(targetArr.map((a)=>search(0,card.length,a)).join(" "));